src/main/java/de/muenchen/dave/services/messstelle/GanglinieService.java

```diff
@@ -0,0 +1,254 @@
+                    seriesEntries.getSeriesEntryGvProzent().getYAxisData().add(intervall.getProzentGueterverkehr());
+                    setLegendInZaehldatenStepline(ladeZaehldatenStepline, ChartLegendUtil.GUETERVERKEHR_ANTEIL_PROZENT);
+                    setRangeMaxPercentInZaehldatenStepline(ladeZaehldatenStepline, intervall.getProzentGueterverkehr());
+
+                    ladeZaehldatenStepline.setXAxisDataFirstChart(
+                            ZaehldatenProcessingUtil.checkAndAddToXAxisWhenNotAvailable(
+                                    ladeZaehldatenStepline.getXAxisDataFirstChart(),
+                                    intervall.getUhrzeitVon().toString()));
+                });
+        ladeZaehldatenStepline.setSeriesEntriesFirstChart(seriesEntries.getChosenStepLineSeriesEntries());
+        return ladeZaehldatenStepline;
+    }
+
+    private static final Integer ROUNDING_VALUE = 20;
+
+    private static final Integer ROUNDING_VALUE_PERCENT = 2;
+
+    protected static void setRangeMaxRoundedToHundredInZaehldatenStepline(final LadeZaehldatenSteplineDTO ladeZaehldatenStepline,
+            final Integer value) {
+        ladeZaehldatenStepline.setRangeMax(
+                ZaehldatenProcessingUtil.getValueRounded(
+                        Math.max(
+                                ZaehldatenProcessingUtil.getZeroIfNull(value),
+                                ladeZaehldatenStepline.getRangeMax()),
+                        ROUNDING_VALUE));
+    }
+
+    protected static void setRangeMaxRoundedToHundredInZaehldatenStepline(final LadeZaehldatenSteplineDTO ladeZaehldatenStepline,
+            final BigDecimal value) {
+        setRangeMaxRoundedToHundredInZaehldatenStepline(
+                ladeZaehldatenStepline,
+                ZaehldatenProcessingUtil.getZeroIfNull(value).intValue());
+    }
+
+    protected static void setRangeMaxPercentInZaehldatenStepline(final LadeZaehldatenSteplineDTO ladeZaehldatenStepline,
+            final BigDecimal value) {
+        final int currentValue = ladeZaehldatenStepline.getRangeMaxPercent();
+        ladeZaehldatenStepline.setRangeMaxPercent(
+                ZaehldatenProcessingUtil.getValueRounded(
+                        BigDecimal.valueOf(currentValue)
+                                .max(ZaehldatenProcessingUtil.getZeroIfNull(value)),
+                        ROUNDING_VALUE_PERCENT));
+    }
+
+    protected static void setLegendInZaehldatenStepline(final LadeZaehldatenSteplineDTO ladeZaehldatenStepline,
+            final String legendEntry) {
+        ladeZaehldatenStepline.setLegend(
+                ChartLegendUtil.checkAndAddToLegendWhenNotAvailable(
+                        ladeZaehldatenStepline.getLegend(),
+                        legendEntry));
+    }
+
+    protected static void setSeriesIndexForFirstChartValue(final StepLineSeriesEntryBaseDTO stepLineSeriesEntry) {
+        stepLineSeriesEntry.setXAxisIndex(0);
```

Comment:
hier und in den unteren zeilen die Magic numbers bitte noch als constanten definieren